tests/test_mailchimp_pagination.py

```diff
@@ -0,0 +1,108 @@
+from math import ceil
+from tap_tester import runner, connections
+from base import MailchimpBaseTest
+
+
+class MailchimpPagination(MailchimpBaseTest):
+    """
+    Ensure tap can replicate multiple pages of data for streams that use pagination.
+    """
+    page_size = 2
+    
+    def name(self):
+        return "tap_tester_mailchimp_pagination_test"
+    
+    def get_properties(self, original=True):
+        props = super().get_properties(original)
+        props['page_size'] = self.page_size
+        return props
+
+    def test_run(self):
+        """
+        • Verify that for each stream you can get multiple pages of data.  
+        This requires we ensure more than 1 page of data exists at all times for any given stream.
+        • Verify by pks that the data replicated matches the data we expect.
+        """
+        
+        # Need to upgrade mailchimp plan for collecting 'automations' stream data. Hence, skipping stream for now. 
+        streams_to_skip = {'automations'}
+        
+        streams_with_2_page_size = {'lists', 'list_segments', 'campaigns', 'unsubscribes'}
+        streams_with_250_page_size = {'list_members', 'list_segment_members'}
+        streams_with_1000_page_size = {'reports_email_activity'}
+        
+        # Verify all the stream are either skipped or tested
+        self.assertEqual(
+            self.expected_check_streams() - streams_to_skip,
+            streams_with_2_page_size | streams_with_250_page_size | streams_with_1000_page_size)
+        
+        self.page_size = 2
```

Comment:
Can we define this in the run_test() method and pass it as an argument in the method?